src/components/Posts/PostsSection.tsx

```diff
@@ -1,41 +1,65 @@
-import { FC } from "react"
-import { map } from "lodash-es"
+import { FC, Fragment } from "react"
+import { isNil, map } from "lodash-es"
+import { Link } from "react-router-dom"
 
 import { usePosts as useUserPosts } from "../../api/requests"
+import { Header3 } from "../text"
+import Divider from "../Divider"
+import SubscribeButton from "../SubscribeButton"
+import { useWhoami } from "../../api/users/requests"
+import Button from "../Button"
 
-import Post from "./Post"
-import PostForm from "./PostForm"
-
+import { PostForm, Post } from "./PostForm"
 
 type PostsSectionProps = {
   door: number
   usePosts?: typeof useUserPosts
   withoutInput?: boolean
 }
 
-const PostsSection: FC<PostsSectionProps> = ({ door, usePosts = useUserPosts, withoutInput = false }) => {
+const PostsSection: FC<PostsSectionProps> = ({
+  door,
+  usePosts = useUserPosts,
+  withoutInput = false
+}) => {
   const { data: posts } = usePosts(door)
+  const { data: whoami } = useWhoami()
 
   if (!posts) return null
 
   return (
-    <section className="mx-4 md:mx-16 lg:w-4/5 lg:mx-auto grid gap-door-elements justify-items-center">
-      {!withoutInput && <PostForm door={door} />}
-
-      {
-        /*
-        * Having min-w-0 (min-width: 0) prevents the content of the grid cells from growing outside of their cell:
-        * https://stackoverflow.com/questions/43311943/prevent-content-from-expanding-grid-items
-        */
-      }
-      <div className="space-y-8 w-full min-w-0">
-        {map(posts, (post) =>
-          <Post
-            key={post.uuid}
-            door={door}
-            post={post}
-          />
-        )}
+    <section className="mx-3 md:mx-16 mt-30 lg:w-full lg:mx-auto grid justify-items-center">
+      <div className="w-full flex flex-col gap-6">
+        <div className="grid grid-cols-[1fr_auto_1fr]">
+          <div />
+          <Header3 as="h2" className="text-center flex-1">
+            {`Kommentarer (${posts.length})`}
+          </Header3>
+          <SubscribeButton door={door} className="place-self-center-end" />
+        </div>
+        <Divider bgClasses="w-full bg-purple-500" />
+      </div>
+      {isNil(whoami?.username) ? (
+        <div className="grid w-192 place-content-center space-y-8 rounded-md bg-purple-500 px-16 py-8">
+          <p className="text-center">
+            Du må oppgi et brukernavn for å kunne delta i kommentarfeltet.
+          </p>
+          <Link className="mx-auto" to="/users/edit">
+            <Button content="Rediger bruker" />
+          </Link>
+        </div>
+      ) : !withoutInput ? (
+        <div className="flex gap-6 mt-25 items-start w-full overflow-hidden">
+          <PostForm door={door} />
+        </div>
+      ) : null}
```

Comment:
Nested ternary med negation i begge steg, og siste term er bare null..?? Ser ut som noe du har kopiert, men for the love of Baby Jesus kan du fikse opp?